bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/ReadEntryProcessorV3.java

```diff
@@ -249,6 +249,12 @@ protected ReadResponse getReadResponse() {
     public void safeRun() {
         requestProcessor.getRequestStats().getReadEntrySchedulingDelayStats().registerSuccessfulEvent(
             MathUtils.elapsedNanos(enqueueNanos), TimeUnit.NANOSECONDS);
+        if (!channel.isOpen()) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Dropping read request for closed channel: {}", channel);
+            }
+            return;
```

Comment:
should it call `requestProcessor.onReadRequestFinish();` before return?

Reply:
Yes, great catch! Thank you. I definitely should have caught that since I just fixed the same bug last week.